st2api/st2api/controllers/v1/packviews.py

```diff
@@ -98,6 +98,9 @@ def get_one(self, name_or_id, *file_path_components):
         """
         pack_db = self._get_by_name_or_id(name_or_id=name_or_id)
 
+        if not pack_db:
+            return abort(404)
```

Comment:
The pattern we tend to use in the API layer is throwing `StackStormDBObjectNotFoundError` exception with an appropriate error message ("Pack with name_or_id foo doesn't exist"). Same goes for other errors (e.g. ValueError will result in 400 response).

This exception is then correctly mapped to 404 inside the JSONErrorResponse hook. Using this approach means error translation is centralized in the hook and it's easy to change the response message or similar (we just need to update one place in the hook).